src/main/java/com/acrolinx/client/sdk/internal/SignInResponse.java

```diff
@@ -18,13 +18,9 @@ private SignInResponse()
     {
     }
 
-    static final TypeSelector<SignInResponse> TYPE_SELECTOR = new TypeSelector<SignInResponse>() {
-        @Override
-        public Class<? extends SignInResponse> getClassForElement(JsonElement readElement)
-        {
-            boolean isSuccess = readElement.getAsJsonObject().get("data").getAsJsonObject().has("accessToken");
-            return isSuccess ? SignInResponse.Success.class : SignInResponse.SignInLinks.class;
-        }
+    static final TypeSelector<SignInResponse> TYPE_SELECTOR = readElement -> {
```

Comment:
Lambdas over anonymous classes for readability and reducing boilerplate code. Let the compiler work for you 🙃.